### ISSUE TITLE
Lag when switching between large files

### ISSUE THREAD

**Issue by forthrin** [derickc/Fountainhead#19]
When switching to a `.fountain` file which has several thousand lines (not uncommon for a screenplay), there is a noticeable delay of about a second when switching to the tab containing the screenplay.

Can performance please be improved so switching is instant for any reasonable screenplay length (up to 10.000 lines which corresponds to 3 hours of film.)

**Comment by derickc** [comment]
Is the problem that you have more than one screenplay open and switching between the two (or more) causes a delay? This is caused by the program scanning for all scenes and characters to create the autocompletions for them. I can look at ways of speeding this up. Currently, there is one file that is overwritten every time a new screenplay is loaded. I did think about creating new files based on screenplay name, but didn't know if people wanted to deal with a growing list of text files that had to be cleaned up. Truthfully, with the size of hard drives now, these text files take up a negligible amount of space.

Is the problem that you only have one screenplay open and when you switch from another Sublime Text file, and switch back to the screenplay there is a delay? This might actually be related to the same issue as above, but I'd have to research it more.

**Comment by forthrin** [comment]
Yes. The delay happens when I switch from a small plain text file to a `.fountain` file with thousands of lines. The delay is at least one second, which is quite noticeable and disruptive. Hope you can speed it up. If this will take long to fix, could you quickly add a way of switching autocompletion off? (This sounds like a non-essential feature, if what it does is give the user tab completion on names of characters.)

**Comment by derickc** [comment]
In the settings section you can turn off character and scene autocompletions. You can also turn off CONT'D.

```
    "characters": false,
    "scenes": false,
    "contd": false,
```

**Comment by forthrin** [comment]
`"characters": false` seems to remove the delay completely. However, I could still press TAB on a character name like "BI" to complete it to "BILL". But it also completed to "Bicycle", "Bifocal", etc. So maybe this is another completion feature?

**Comment by derickc** [comment]
You can turn autocomplete off, and also make tab not perform autocompletions and insert a normal tab.

```
"auto_complete": false,
"tab_completion": false,
```

**Comment by forthrin** [comment]
Will reopen if required
